src/content/changelogs/durable-objects.yaml

```
@@ -5,6 +5,9 @@ productLink: "/durable-objects/"
 productArea: Developer platform
 productAreaLink: /workers/platform/changelog/platform/
 entries:
+  - publish_date: "2025-02-11"
+    description: |-
+      - When the runtime generates an "internal error" exception in response to certain failures, the exception message may include a reference ID that customers can include in support communication, to help make the error easier to diagnose.  For example, an exception with the new message might look like: `internal error; reference = 0123456789abcdefghijklmn`.
```

```suggestion
      - When Durable Objects generate an "internal error" exception in response to certain failures, the exception message may include a reference ID that customers can include in support communication, to help make the error easier to diagnose.  For example, an exception with the new message might look like: `internal error; reference = 0123456789abcdefghijklmn`.
```

@jclee Why `may include`? Are there internal errors that won't have a reference ID?

---

I think as currently implemented, most exceptions that were generated with the text "internal error" will have the reference id.

There is one remaining "internal error" exception that requires more work to add the reference id, but I think it is only generated in rare circumstances, like worker startup exceptions that should mostly be caught during worker validation.

And there are some other instances where we generate similar but more verbose exceptions -- like `jsg.Error: Internal error in Durable Object storage write caused object to be reset.` -- that we may want to also add reference ids to, but that are not affected by the current implementation.